### PR TITLE
Refactor Code Climate engine options parsing

### PR DESCRIPTION
**Pull request by noahd1**
The code was all in the binary making it a bit harder to test or change easily.

No behavior change is intended in this PR.

(Small change coming in next PR)

### CODE REVIEW

**Comment by presidentbeef** [comment]
Looks nice to me. I like the added testability.

File: bin/codeclimate-brakeman [hunk 1]
```diff
@@ -3,34 +3,19 @@ $:.unshift "#{File.expand_path(File.dirname(__FILE__))}/../lib"
 
 require "brakeman"
 require "json"
+require "brakeman/codeclimate/engine_configuration"
 
-options = {
+brakeman_options = {
```
Comment (presidentbeef) [inline bin/codeclimate-brakeman]: Don't need this now, right?

Comment (noahd1) [reply bin/codeclimate-brakeman]: oops, good call.

**Comment by noahd1** [comment]
Addressed feedback - also fixed the order of args to `assert_equal`

**Comment by noahd1** [comment]
This is a pre-requisite for a soon to follow [PR](https://github.com/noahd1/brakeman/pull/1).

**Pull request merged**